test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/DefaultHtmlGeneratorTest.cs

```diff
@@ -676,8 +677,58 @@ public void GenerateAntiforgery_AlwaysGeneratesAntiforgeryToken_IfCannotRenderAt
             Assert.Equal(expected, antiforgeryField);
         }
 
+        [Fact]
+        public async Task RenderPartialViewAsync_Throws_IfViewCannotBeFound()
+        {
+            // Arrange
+            var expected = "The partial view 'test-view' was not found. The following locations were searched:" +
+                Environment.NewLine +
+                "location1" + Environment.NewLine +
+                "location2" + Environment.NewLine +
+                "location3" + Environment.NewLine +
+                "location4";
+
+            var viewEngine = new Mock<ICompositeViewEngine>(MockBehavior.Strict);
+            viewEngine
+                .Setup(v => v.GetView(/*executingFilePath*/ null, It.IsAny<string>(), /*isMainPage*/ false))
+                .Returns(ViewEngineResult.NotFound("test-view", new[] { "location1", "location2" }))
+                .Verifiable();
+            viewEngine
+                .Setup(v => v.FindView(It.IsAny<ActionContext>(), It.IsAny<string>(), /*isMainPage*/ false))
+                .Returns(ViewEngineResult.NotFound("test-view", new[] { "location3", "location4" }))
```

Comment:
I assume we have a test somewhere else where `GetView` fails but `FindView` succeeds? If not, it might be worth adding one.